sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/kafka/AzureKafkaAutoconfigurationUtils.java

```diff
@@ -0,0 +1,94 @@
+import com.azure.spring.cloud.service.implementation.kafka.KafkaOAuth2AuthenticateCallbackHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.azure.spring.cloud.service.implementation.kafka.AzureKafkaPropertiesUtils.convertConfigMapToAzureProperties;
+import static com.azure.spring.cloud.core.implementation.util.AzurePropertiesUtils.copyPropertiesIgnoreNull;
+import static org.apache.kafka.clients.CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_JAAS_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_MECHANISM;
+import static org.apache.kafka.common.security.auth.SecurityProtocol.SASL_SSL;
+import static org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule.OAUTHBEARER_MECHANISM;
+
+public final class AzureKafkaAutoconfigurationUtils {
+
+    public static final Map<String, String> KAFKA_OAUTH_CONFIGS;
+    public static final String SECURITY_PROTOCOL_CONFIG_SASL = SASL_SSL.name();
+    public static final String SASL_MECHANISM_OAUTH = OAUTHBEARER_MECHANISM;
+    public static final String SASL_JAAS_CONFIG_OAUTH =
+        "org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required;";
+    public static final String SASL_LOGIN_CALLBACK_HANDLER_CLASS_OAUTH =
+        KafkaOAuth2AuthenticateCallbackHandler.class.getName();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureKafkaAutoconfigurationUtils.class);
+    private static final String LOG_PROPERTIES_CONFIGURE = "Property %s will be configured as %s.";
+
+    static {
+        Map<String, String> configs = new HashMap<>();
+        configs.put(SECURITY_PROTOCOL_CONFIG, SECURITY_PROTOCOL_CONFIG_SASL);
+        configs.put(SASL_MECHANISM, SASL_MECHANISM_OAUTH);
+        configs.put(SASL_JAAS_CONFIG, SASL_JAAS_CONFIG_OAUTH);
+        configs.put(SASL_LOGIN_CALLBACK_HANDLER_CLASS, SASL_LOGIN_CALLBACK_HANDLER_CLASS_OAUTH);
+        KAFKA_OAUTH_CONFIGS = Collections.unmodifiableMap(configs);
+    }
+
+    private AzureKafkaAutoconfigurationUtils() {
+    }
+
+    /**
+     * Detect whether we need to configure SASL/OAUTHBEARER properties for {@link KafkaProperties}. Will configure when
+     * the security protocol is not configured, or it's set as SASL_SSL with sasl mechanism as null or OAUTHBEAR.
+     *
+     * @param sourceProperties the source kafka properties for admin/consumer/producer to detect
+     * @return whether we need to configure with Spring Cloud Azure MSI support or not.
+     */
+    public static boolean needConfigureSaslOAuth(Map<String, Object> sourceProperties) {
+        String securityProtocol = (String) sourceProperties.get(SECURITY_PROTOCOL_CONFIG);
```

Comment:
is there any chance to throw `ClassCastException` here?

Reply:
it should be, the several properties here we want to parese from KafkaProperties have all been converted to String by KafkaProperties. but do you mean  any changes from KafkaProperties in the future?